pkg/hub/managedclusterset/controller.go

```diff
@@ -42,172 +52,216 @@ func NewManagedClusterSetController(
 		clusterLister:    clusterInformer.Lister(),
 		clusterSetLister: clusterSetInformer.Lister(),
 		eventRecorder:    recorder.WithComponentSuffix("managed-cluster-set-controller"),
+
+		clusterSetsMap: map[string]string{},
 	}
 	return factory.New().
 		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
 			accessor, _ := meta.Accessor(obj)
-			return accessor.GetName()
+			return fmt.Sprintf("%s%s", clusterSetKeyPrefix, accessor.GetName())
 		}, clusterSetInformer.Informer()).
 		// TODO, create a PR for library-go to accept a queue key function, which
 		// is able to produce a []string instead of string based on one object, when
 		// registering event handler. And then refactor the logic here.
-		WithInformers(clusterInformer.Informer()).
+		WithInformersQueueKeyFunc(func(obj runtime.Object) string {
+			accessor, _ := meta.Accessor(obj)
```

Comment:
could i suggest we have two QueueKeyFunc for cluster Informers. One to enqueue the previous clusterset, one to enqueu the current clusterset?  instead of having a special prefix?

Reply:
Code refactored as suggested.